backend/tests/Feature/StepTypedOutputsTest.php

```
@@ -0,0 +1,255 @@
+    }
+
+    public function test_recording_is_idempotent_overwrite(): void
+    {
+        $output = $this->def(['value_type' => 'number']);
+        $url = "/operator/batch-step/{$this->batchStep->id}/outputs/{$output->id}";
+
+        $this->asOperator()->post($url, ['value' => '1'])->assertRedirect();
+        $this->asOperator()->post($url, ['value' => '2'])->assertRedirect();
+
+        // One live value (the latest); the prior is soft-deleted (audit).
+        $this->assertSame(1, BatchStepOutputValue::where('output_id', $output->id)->count());
```

PHPStan (2.2.7)

[error] Call to an undefined static method App\Models\BatchStepOutputValue::where().


(staticMethod.notFound)